edalize/symbiflow.py

```diff
@@ -0,0 +1,160 @@
+import logging
+import os.path
+import platform
+import re
+import subprocess
+
+from edalize.edatool import Edatool
+from edalize.yosys import Yosys
+from importlib import import_module
+
+logger = logging.getLogger(__name__)
+
+""" Symbiflow backtend
+
+A core (usually the system core) can add the following files:
+
+- Standard design sources (Verilog only)
+
+- Constraints: unmanaged constraints with file_type SDC, pin_constraints with file_type PCF and placement constraints with file_type xdc
+
+"""
+
+
+class Symbiflow(Edatool):
+
+    argtypes = ["vlogdefine", "vlogparam", "generic"]
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            symbiflow_help = {
+                "members": [
+                    {
+                        "name": "package",
+                        "type": "String",
+                        "desc": "FPGA chip package (e.g. clg400-1)",
+                    },
+                    {
+                        "name": "part",
+                        "type": "String",
+                        "desc": "FPGA part type (e.g. xc7a50t)",
+                    },
+                    {
+                        "name": "builddir",
+                        "type": "String",
+                        "desc": 'directory where all the intermediate files will be stored (default "build")',
+                    },
+                    {
+                        "name": "vendor",
+                        "type": "String",
+                        "desc": 'Target architecture. Currently only "xilinx" is supported',
+                    },
+                    {
+                        "name": "pnr",
+                        "type": "String",
+                        "desc": 'Place and Route tool. Currently only "vpr" is supported',
+                    },
+                    {
+                        "name": "options",
```

Comment:
Can we name this one vpr_options instead to be consistent with other backends?

Reply:
Changed name to ``additional_vpr_options``.

Reply:
I see this is only unresolved comment.

I named this option as ``additional_vpr_options`` to emphasize that, this options, are added to the default one defined in toolchain_wrapper. 
Currently there is no option to overwrite ``vpr_options`` from toolchain_wrapper command-line argument, as for now, there is no real usage for it, but in the future we are planning to add this option and update this backend with ``vpr_options`` that will overwrite default options.

@olofk What do you think about it? Is missing ``vpr_options`` is a blocker for this PR and I should add this now?

Reply:
Sorry. The real issue was that I couldn't find time to finish the review. But yes, I would prefer to just call it vpr_options. Ideally, the wrapper should only set values that it absolutely has to set and leave the rest up to the user

Reply:
Ok, done.